tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala

```diff
@@ -29,13 +37,76 @@ class Tarmac(val root: Configurable) extends QScript with PedigreeQscript with S
 
   class Sample(name: String) extends AbstractSample(name) {
 
-    val inputCountFile: Option[File] = config("count_file")
+    val inputXhmmCountFile: Option[File] = config("xhmm_count_file")
+    val inputWisecondorCountFile: Option[File] = config("wisecondor_count_file")
     val bamFile: Option[File] = config("bam")
 
+    /**
+     * Create XHMM count file or create link to input count file
+     * Precedence is given to existing count files.
+     * Returns a disjunction where right is the file, and left is
+     * a potential error message
+     */
+    lazy val outputXhmmCountFile: String \/ File = {
+      val outFile = new File(sampleDir + File.separator + s"$name.dcov")
+      (inputXhmmCountFile, bamFile) match {
+        case (Some(f), _) => {
+          val ln = new Ln(root)
+          ln.input = f
+          ln.output = outFile
+          add(ln)
+          \/-(ln.output)
+        }
+        case (None, Some(bam)) => {
+          val dcov = DepthOfCoverage(root, List(bam), outFile, List(targets))
+          add(dcov)
+          \/-(dcov.out)
+        }
+        case _ => -\/(s"Cannot find bam file or xhmm count file for sample" +
+          s" $name in config. At least one must be given.")
+      }
+
+    }
+
+    /**
+     * Create wisecondor count file or create link to input count file.
+     * Precedence is given to existing count files.
+     * Returns a disjunction where right is the file, and left is
+     * a potential error message
+     */
+    lazy val outputWisecondorCountFile: String \/ File = {
+      val outFile = new File(sampleDir + File.separator + s"$name.wisecondor.bed")
+      (inputWisecondorCountFile, bamFile) match {
+        case (Some(f), _) => {
+          val ln = new Ln(root)
+          ln.input = f
+          ln.output = outFile
+          add(ln)
+          \/-(ln.output)
+        }
+        case (None, Some(bam)) => {
+          val counter = new WisecondorCount(root)
+          counter.inputBam = bam
+          counter.output = outFile
+          counter.binFile = Some(targets)
+          add(counter)
+          \/-(counter.output)
+        }
+        case _ => -\/(s"Cannot find bam file or wisecondor count for sample" +
+          s" $name. At least one must be given.")
+      }
+
+    }
+
     /** Function to add sample jobs */
-    def addJobs(): Unit = {}
+    def addJobs(): Unit = {
```

Comment:
You can also just make `outputWisecondorCountFile` just create the job object instead of adding it. It is `addJobs` who would then add the output, so you have one single place where all the side-effecting function (the job addition function) is done.